test/integration/targets/azure_rm_virtualmachine/basic.yaml

```diff
@@ -0,0 +1,291 @@
+- name: Create virtual machine with a single NIC and legacy storage
+  azure_rm_virtualmachine:
+    resource_group: "{{ hostvars['localhost'].resource_group }}"
+    name: "{{ hostvars['localhost'].vm_names.basic }}"
+    short_hostname: "{{ hostvars['localhost'].vm_names.basic }}"
+    admin_username: "{{ hostvars['localhost'].admin_user }}"
+    admin_password: "{{ hostvars['localhost'].password }}"
+    availability_set: "{{ hostvars['localhost'].availability_set_blob }}"
+    network_interface_names: "{{ hostvars['localhost'].vm_nics[0] }}"
+    storage_account: "{{ hostvars['localhost'].storage_account }}"
+    storage_container: "{{ hostvars['localhost'].vm_names.basic }}"
+    storage_blob: "{{ hostvars['localhost'].vm_names.basic }}.vhd"
+    os_type: "{{ hostvars['localhost'].os_type }}"
+    image: "{{ hostvars['localhost'].image }}"
+    vm_size: "{{ hostvars['localhost'].vm_size }}"
+    custom_data: |
+      #!/bin/sh
+      echo "custom_data was executed" > /tmp/custom_data.txt
+  register: output
+
+- assert:
+    that:
+      - azure_vm.properties.availabilitySet.id
+
+- name: Should be idempotent with a single NIC and legacy storage
+  azure_rm_virtualmachine:
+    resource_group: "{{ hostvars['localhost'].resource_group }}"
+    name: "{{ hostvars['localhost'].vm_names.basic }}"
+    short_hostname: "{{ hostvars['localhost'].vm_names.basic }}"
+    admin_username: "{{ hostvars['localhost'].admin_user }}"
+    admin_password: "{{ hostvars['localhost'].password }}"
+    availability_set: "{{ hostvars['localhost'].availability_set_blob }}"
+    network_interface_names: "{{ hostvars['localhost'].vm_nics[0] }}"
+    storage_account: "{{ hostvars['localhost'].storage_account }}"
+    storage_container: "{{ hostvars['localhost'].vm_names.basic }}"
+    storage_blob: "{{ hostvars['localhost'].vm_names.basic }}.vhd"
+    os_type: "{{ hostvars['localhost'].os_type }}"
+    image: "{{ hostvars['localhost'].image }}"
+    vm_size: "{{ hostvars['localhost'].vm_size }}"
+  register: output
+
+- assert:
+    that: not output.changed
+
+#- add_host:
+    #name: new_azure_vm
+    #ansible_host: '{{ output.ansible_facts.azure_vm.properties.networkProfile.networkInterfaces[0].properties.ipConfigurations[0].properties.publicIPAddress.properties.ipAddress }}'
+    #ansible_connection: paramiko  # not guaranteed to have sshpass...
+    #ansible_user: "{{ hostvars['localhost'].admin_user }}"
+    #ansible_password: "{{ hostvars['localhost'].password }}"
+    #ansible_host_key_checking: false
+
+- name: wait for SSH port to be open
+  wait_for:
+    host: "{{ output.ansible_facts.azure_vm.properties.networkProfile.networkInterfaces[0].properties.ipConfigurations[0].properties.publicIPAddress.properties.ipAddress }}"
+    port: 22
+    timeout: 60
+    state: started
+
+#- block:
+#   TODO: figure out how to make this work under ansible-test with the coverage injector
```

Comment:
I realize these tests were already disabled in the original tests, but they could be made to work by setting the `ANSIBLE_TEST_REMOTE_INTERPRETER` environment variable to an empty string, or to the path to the interpreter:

```yaml
    environment:
        ANSIBLE_TEST_REMOTE_INTERPRETER: ""
```

This will disable the coverage injector for the delegated hosts.